src/components/Tx/TxHistory/TxEntry.jsx

```diff
@@ -0,0 +1,349 @@
+    let tokensTo
+
+    if (isTokenTransfer) {
+      const { decimals, symbol } = tx.token
+      const tokenCount = tx.params[1].slice(0, -Math.abs(decimals))
+      const tokenSymbol = symbol || 'tokens'
+      description = `Transferred ${tokenCount} ${tokenSymbol}`
+      tokensTo = tx.params[0]
+    } else if (tx.isNewContract) {
+      description = 'Created New Contract'
+    } else if (tx.executionFunction) {
+      const executionFunctionSentence =
+        tx.executionFunction.charAt(0).toUpperCase() +
+        tx.executionFunction
+          .slice(1, tx.executionFunction.indexOf('('))
+          .replace(/([A-Z]+|[0-9]+)/g, ' $1')
+      description = `Executed ${executionFunctionSentence} function`
+    } else {
+      const etherAmount = web3.utils.fromWei(tx.value, 'ether')
+      description = `Sent ${etherAmount} Ether`
+    }
+
+    let status = (
+      <span className="value" style={{ color: 'grey' }}>
+        {i18n.t('mist.txHistory.statusPending')}
+      </span>
+    )
+    if (tx.status === 0) {
+      status = (
+        <span className="value" style={{ color: 'red' }}>
+          {i18n.t('mist.txHistory.statusFailed')}
+        </span>
+      )
+    } else if (tx.status === 1 && tx.blockNumber) {
+      const { blockNumber } = this.props
+      const numberConfirmations = blockNumber - tx.blockNumber
+
+      status = (
+        <span>
+          <span className="value" style={{ color: 'green' }}>
+            {i18n.t('mist.txHistory.statusConfirmed')}
+          </span>{' '}
+          <span>
+            ({numberConfirmations} {i18n.t('mist.txHistory.confirmations')})
+          </span>
+        </span>
+      )
+    }
+
+    return (
+      <StyledTx key={tx.hash || tx.nonce}>
+        <div className="right">
+          {network && <div className="network">{network}</div>}
+          <div className="date">{date}</div>
+        </div>
+        <div className="description">{description}</div>
+        {tx.contractAddress && (
+          <div>
+            <div className="label">{i18n.t('mist.txHistory.newContract')}</div>
+            <div className="identicon">
+              <Identicon address={tx.contractAddress} size="small" />
+            </div>
+            <span className="value">{tx.contractAddress}</span>
+          </div>
+        )}
+        <div>
+          <div className="label">{i18n.t('mist.txHistory.from')}</div>
+          <div className="identicon">
+            <Identicon address={tx.from} size="small" />
+          </div>
+          <span className="value">{tx.from}</span>
+        </div>
+        {isTokenTransfer && (
+          <div>
+            <div className="label">{i18n.t('mist.txHistory.to')}</div>
+            <div className="identicon">
+              <Identicon address={tokensTo} size="small" />
+            </div>
+            <span className="value">{tokensTo}</span>
+          </div>
+        )}
+        {!isTokenTransfer && (
+          <div>
+            {tx.to && (
```

Comment:
I think this component can be broken down into a few sub-components. It's a bit unfriendly to read, because of all the conditional logic forks. Consider each fork as a potential child component refactor. Could be a separate PR if you prefer.

Reply:
Did a good amount of reorg and simplification, should be much easier to read the component now. Thanks!